### PR TITLE
Create new pins_arduino.h and update boards.txt to add Heltec Wireless Tracker board

### PR DESCRIPTION
**Pull request by jhiggason**
-----------
## Description of Change
Add Heltec Wireless Tracker to variants and boards.txt using https://github.com/Heltec-Aaron-Lee/WiFi_Kit_series/releases as a source.  

This is my first time doing this- not sure if I need to include the libraries they have or not.  (GNSS and LORA are the two that come to mind) 

## Tests scenarios
Please describe on what Hardware and Software combinations you have tested this Pull Request and how.

I'm not 100% sure how to test this but happy to do it since I have a board.  

## Related links
https://github.com/espressif/arduino-esp32/pull/8464
https://github.com/platformio/platform-espressif32/issues/1148

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -15819,6 +15819,253 @@ heltec_wireless_stick_lite.menu.EraseFlash.none.upload.erase_cmd=
...
+
+heltec_wireless_tracker.menu.LORAWAN_PREAMBLE_LENGTH.0=8(default)
+heltec_wireless_tracker.menu.LORAWAN_PREAMBLE_LENGTH.0.build.LORAWAN_PREAMBLE_LENGTH=8
+heltec_wireless_tracker.menu.LORAWAN_PREAMBLE_LENGTH.1=16(For M00 and M00L)
+heltec_wireless_tracker.menu.LORAWAN_PREAMBLE_LENGTH.1.build.LORAWAN_PREAMBLE_LENGTH=16
+
+heltec_wireless_tracker.build.defines=-D{build.band} {build.psram_val} -DLoRaWAN_DEBUG_LEVEL={build.LoRaWanDebugLevel} -DACTIVE_REGION=LORAMAC_{build.band} -DLORAWAN_PREAMBLE_LENGTH={build.LORAWAN_PREAMBLE_LENGTH} -DLORAWAN_DEVEUI_AUTO={build.LORAWAN_DEVEUI_AUTO} -D{build.board}
+heltec_wireless_tracker.build.extra_libs=-lheltec_{build.cdc_on_boot}
+
+
```
Comment (P-R-O-C-H-Y) [inline boards.txt:16068]: Please add missing EraseFlash menu.
```suggestion
heltec_wireless_tracker.menu.EraseFlash.none=Disabled
heltec_wireless_tracker.menu.EraseFlash.none.upload.erase_cmd=
heltec_wireless_tracker.menu.EraseFlash.all=Enabled
heltec_wireless_tracker.menu.EraseFlash.all.upload.erase_cmd=-e

```

File: variants/heltec_wireless_tracker/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+#include <stdint.h>
+#include "soc/soc_caps.h"
+
+#define USB_VID 0x303a
+#define USB_PID 0x1001
+
+#define EXTERNAL_NUM_INTERRUPTS 46
+#define NUM_DIGITAL_PINS        48
+#define NUM_ANALOG_INPUTS       20
```
Comment (P-R-O-C-H-Y) [inline variants/heltec_wireless_tracker/pins_arduino.h:12]: This is no longer needed and defined globally for all SOCs in Arduino.h file.
```suggestion
```

File: variants/heltec_wireless_tracker/pins_arduino.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+// and change this setup for the chosen pin (for example 38)
+static const uint8_t LED_BUILTIN = SOC_GPIO_PIN_COUNT+48;
+#define BUILTIN_LED  LED_BUILTIN // backward compatibility
+#define LED_BUILTIN LED_BUILTIN
+#define RGB_BUILTIN LED_BUILTIN
+#define RGB_BRIGHTNESS 64
+
+#define analogInputToDigitalPin(p)  (((p)<20)?(analogChannelToDigitalPin(p)):-1)
+#define digitalPinToInterrupt(p)    (((p)<48)?(p):-1)
+#define digitalPinHasPWM(p)         (p < 46)
```
Comment (P-R-O-C-H-Y) [inline variants/heltec_wireless_tracker/pins_arduino.h:25]: This is no longer needed and defined globally for all SOCs in Arduino.h file.
```suggestion
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
Hi @jhiggason, please take a look on my comment.
Added suggestions for easier fixing.

**Comment by jhiggason** [comment]
@P-R-O-C-H-Y  I made the requested changes- can you check them out for me?  Do you see anything else wrong with it?

**Comment by P-R-O-C-H-Y** [comment]
@jhiggason Hi, you did not followed up the suggestion or required change for the erase flash menu. Can you fix that and place it on correct place of the board definition (the end, like all other boards have it)?

File: boards.txt [hunk 4]
```diff
@@ -15986,6 +15986,256 @@ heltec_wireless_stick_lite.menu.EraseFlash.none.upload.erase_cmd=
...
+heltec_wireless_tracker.menu.FlashMode.dio.build.flash_freq=80m
+heltec_wireless_tracker.menu.FlashMode.opi=OPI 80MHz
+heltec_wireless_tracker.menu.FlashMode.opi.build.flash_mode=dout
+heltec_wireless_tracker.menu.FlashMode.opi.build.boot=opi
+heltec_wireless_tracker.menu.FlashMode.opi.build.boot_freq=80m
+heltec_wireless_tracker.menu.FlashMode.opi.build.flash_freq=80m
+heltec_wireless_tracker.build.defines=-D{build.band} {build.psram_val} -DLoRaWAN_DEBUG_LEVEL={build.LoRaWanDebugLevel} -DACTIVE_REGION=LORAMAC_{build.band} -DLORAWAN_PREAMBLE_LENGTH={build.LORAWAN_PREAMBLE_LENGTH} -DLORAWAN_DEVEUI_AUTO={build.LORAWAN_DEVEUI_AUTO} -D{build.board}
+heltec_wireless_tracker.build.extra_libs=-lheltec_{build.cdc_on_boot}
+
+heltec_wireless_tracker.menu.EraseFlash.none=Disabled
```
Comment (P-R-O-C-H-Y) [inline boards.txt:16069]: build.board.defines already exist at the end of board definition.
Erase flash menu is not complete and please add it to the end of board definition as other boards do (check next comment)
```suggestion

```

File: boards.txt [hunk 5]
```diff
@@ -15986,6 +15986,256 @@ heltec_wireless_stick_lite.menu.EraseFlash.none.upload.erase_cmd=
...
+
+heltec_wireless_tracker.menu.LORAWAN_PREAMBLE_LENGTH.0=8(default)
+heltec_wireless_tracker.menu.LORAWAN_PREAMBLE_LENGTH.0.build.LORAWAN_PREAMBLE_LENGTH=8
+heltec_wireless_tracker.menu.LORAWAN_PREAMBLE_LENGTH.1=16(For M00 and M00L)
+heltec_wireless_tracker.menu.LORAWAN_PREAMBLE_LENGTH.1.build.LORAWAN_PREAMBLE_LENGTH=16
+
+heltec_wireless_tracker.build.defines=-D{build.band} {build.psram_val} -DLoRaWAN_DEBUG_LEVEL={build.LoRaWanDebugLevel} -DACTIVE_REGION=LORAMAC_{build.band} -DLORAWAN_PREAMBLE_LENGTH={build.LORAWAN_PREAMBLE_LENGTH} -DLORAWAN_DEVEUI_AUTO={build.LORAWAN_DEVEUI_AUTO} -D{build.board}
+heltec_wireless_tracker.build.extra_libs=-lheltec_{build.cdc_on_boot}
+
+
```
Comment (P-R-O-C-H-Y) [inline boards.txt:16238]:
```suggestion
heltec_wireless_tracker.menu.EraseFlash.none=Disabled
heltec_wireless_tracker.menu.EraseFlash.none.upload.erase_cmd=
heltec_wireless_tracker.menu.EraseFlash.all=Enabled
heltec_wireless_tracker.menu.EraseFlash.all.upload.erase_cmd=-e

```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@jhiggason PTAL

File: boards.txt (hunk 5 above)
Comment (jhiggason) [reply boards.txt:16238]: Added this section.

**Comment by jhiggason** [comment]
@P-R-O-C-H-Y Ok let's try again.  Is there a way for me to build this and test it before we merge it into master?  Could you point me to a doc if yes?  (Sorry, I'm just starting my programming journey with ESP32's and some of this, like source building and library control is new to me)  Thank you for your help with this- I really do appreciate it.

File: boards.txt [hunk 6]
```diff
@@ -15988,6 +15988,257 @@ heltec_wireless_stick_lite.menu.EraseFlash.all.upload.erase_cmd=-e
...
+heltec_wireless_tracker.menu.FlashMode.dio.build.flash_freq=80m
+heltec_wireless_tracker.menu.FlashMode.opi=OPI 80MHz
+heltec_wireless_tracker.menu.FlashMode.opi.build.flash_mode=dout
+heltec_wireless_tracker.menu.FlashMode.opi.build.boot=opi
+heltec_wireless_tracker.menu.FlashMode.opi.build.boot_freq=80m
+heltec_wireless_tracker.menu.FlashMode.opi.build.flash_freq=80m
+heltec_wireless_tracker.build.defines=-D{build.band} {build.psram_val} -DLoRaWAN_DEBUG_LEVEL={build.LoRaWanDebugLevel} -DACTIVE_REGION=LORAMAC_{build.band} -DLORAWAN_PREAMBLE_LENGTH={build.LORAWAN_PREAMBLE_LENGTH} -DLORAWAN_DEVEUI_AUTO={build.LORAWAN_DEVEUI_AUTO} -D{build.board}
+heltec_wireless_tracker.build.extra_libs=-lheltec_{build.cdc_on_boot}
+
+heltec_wireless_tracker.menu.EraseFlash.none=Disabled
```
Comment (P-R-O-C-H-Y) [inline boards.txt:16069]: Please remove this line as it was added to the end of the file.
```suggestion
```

File: boards.txt [hunk 7]
```diff
@@ -15988,6 +15988,257 @@ heltec_wireless_stick_lite.menu.EraseFlash.all.upload.erase_cmd=-e
...
+heltec_wireless_tracker.menu.FlashMode.dio.build.boot=dio
+heltec_wireless_tracker.menu.FlashMode.dio.build.boot_freq=80m
+heltec_wireless_tracker.menu.FlashMode.dio.build.flash_freq=80m
+heltec_wireless_tracker.menu.FlashMode.opi=OPI 80MHz
+heltec_wireless_tracker.menu.FlashMode.opi.build.flash_mode=dout
+heltec_wireless_tracker.menu.FlashMode.opi.build.boot=opi
+heltec_wireless_tracker.menu.FlashMode.opi.build.boot_freq=80m
+heltec_wireless_tracker.menu.FlashMode.opi.build.flash_freq=80m
+heltec_wireless_tracker.build.defines=-D{build.band} {build.psram_val} -DLoRaWAN_DEBUG_LEVEL={build.LoRaWanDebugLevel} -DACTIVE_REGION=LORAMAC_{build.band} -DLORAWAN_PREAMBLE_LENGTH={build.LORAWAN_PREAMBLE_LENGTH} -DLORAWAN_DEVEUI_AUTO={build.LORAWAN_DEVEUI_AUTO} -D{build.board}
+heltec_wireless_tracker.build.extra_libs=-lheltec_{build.cdc_on_boot}
```
Comment (P-R-O-C-H-Y) [inline boards.txt:16067]: Can you please come this part to the end of the board definition just right above EraseFlash menu?

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]

**Comment by P-R-O-C-H-Y** [comment]
> @P-R-O-C-H-Y Ok let's try again. Is there a way for me to build this and test it before we merge it into master? Could you point me to a doc if yes? (Sorry, I'm just starting my programming journey with ESP32's and some of this, like source building and library control is new to me) Thank you for your help with this- I really do appreciate it.

@jhiggason Yes you can build na test the board configuration. You can just add the changes you are adding to the ESP32 core. 
If you install ESP32 Arduino Core through Boards manager in Arduino IDE, you can find the location of esp32 by going to Arduino15 folder located here:
```
   Windows: C:\Users\{username}\AppData\Local\Arduino15
   macOS: /Users/{username}/Library/Arduino15
   Linux: home/{username}/.arduino15
```
and than go to `/packages/esp32/hardware/esp32/<version>/...` and there is the boards.txt file and variants folder you are editing :).

If you Installed manually from GitHub, it is located in `Documents/Arduino/hardware/`.
After you add the changes there, you may need to restart the Arduino IDE to see the new board you added (better to use old IDE 1.8.x).

**Comment by VojtechBartoska** [comment]
any updates on this @jhiggason?

**Comment by jhiggason** [comment]
Sorry!  Completely forgot about this-  I'll try to test and get back to you today.

**Comment by Lapporatory992** [comment]
Hi,
Would it possible to add the HeltecESP32 "wireless_paper" board to the list of Espressif supported boards ?
https://github.com/Heltec-Aaron-Lee/WiFi_Kit_series/blob/master/esp32/boards.txt

Thank you,

**Comment by songpola** [comment]
@jhiggason Hello! Just to let you know that they just released 1.0.0 release. :D

**Pull request merged**